### PR TITLE
feat: add login function

### PR DESCRIPTION
**Pull request by vin-oys**

### CODE REVIEW

File: api/user.go [hunk 1]
```diff
@@ -45,11 +78,19 @@ func (server *Server) createUser(ctx *gin.Context) {
 
 	user, err := server.store.CreateUser(ctx, arg)
 	if err != nil {
+		if pqErr, ok := err.(*pq.Error); ok {
+			switch pqErr.Code.Name() {
+			case "unique_violation":
```
Comment (weehongayden) [inline api/user.go:83]: The alignment here need to fix

File: api/user.go [hunk 2]
```diff
@@ -65,7 +106,8 @@ func (server *Server) getUser(ctx *gin.Context) {
 		return
 	}
 
-	ctx.JSON(http.StatusOK, user)
+	rsp := newUserResponse(user)
```
Comment (weehongayden) [inline api/user.go:109]: Is that possible to merge lines 109 and 110 into one line?

File: api/user.go [hunk 3]
```diff
@@ -80,13 +122,14 @@ func (server *Server) updateUser(ctx *gin.Context) {
...
 
-	res, err := server.store.UpdateUser(ctx, arg)
+	user, err := server.store.UpdateUser(ctx, arg)
 	if err != nil {
 		ctx.JSON(http.StatusInternalServerError, errorResponse(err))
 		return
 	}
 
-	ctx.JSON(http.StatusOK, res)
+	rsp := newUserResponse(user)
```
Comment (weehongayden) [inline api/user.go:131]: Is that possible to merge lines 131 and 132 into one line?

File: api/user.go (hunk 1 above)
Comment (vin-oys) [reply api/user.go:83]: its IDE auto format

**Review by weehongayden — APPROVED** [review]

**Pull request merged**